src/Checkbox/CheckboxComponent.jsx

```diff
@@ -13,10 +13,16 @@ export default observer(() => {
         id="example-checkbox"
         name="foo"
         onChange={(event) => {
-          checkboxPresenter.setIsChecked(event.target.checked)
+          dispatch(
+            {type: 'SET_IS_CHECKED', payload: event.target.checked}
+          );
         }}
-        checked={checkboxPresenter.isChecked}
+        checked={state.isChecked}
       />
+      { checkedAndUnchecked(state)
+        ? <p>This was checked and then unchecked</p>
+        : null
+      }
```

Comment:
`checkedAndUnchecked()` is like a computed value in [MobX](https://mobx.js.org/computeds-with-args.html) and [Vue.js](https://v2.vuejs.org/v2/guide/computed.html).

It's derived state from other state.

But here, all you have to do is call `checkedAndUnchecked(state)`.

That's a pure function, so it's really easy to test, and no need to mark it `@computed`